ambari-server/src/main/python/ambari-server.py

```diff
@@ -600,11 +601,20 @@ def init_setup_sso_options(parser):
   parser.add_option('--ambari-admin-username', default=None, help="Ambari administrator username for accessing Ambari's REST API", dest="ambari_admin_username")
   parser.add_option('--ambari-admin-password', default=None, help="Ambari administrator password for accessing Ambari's REST API", dest="ambari_admin_password")
 
+
 @OsFamilyFuncImpl(OsFamilyImpl.DEFAULT)
 def init_pam_setup_parser_options(parser):
   parser.add_option('--pam-config-file', default=None, help="Path to the PAM configuration file", dest="pam_config_file")
   parser.add_option('--pam-auto-create-groups', default=None, help="Automatically create groups for authenticated users [true/false]", dest="pam_auto_create_groups")
 
+
+@OsFamilyFuncImpl(OsFamilyImpl.DEFAULT)
+def init_tproxy_setup_parser_options(parser):
+  parser.add_option('--ambari-admin-username', default=None, help="Ambari administrator username for accessing Ambari's REST API", dest="ambari_admin_username")
+  parser.add_option('--ambari-admin-password', default=None, help="Ambari administrator password for accessing Ambari's REST API", dest="ambari_admin_password")
+  parser.add_option('--tproxy-enabled', default=None, help="Indicates whether to enable/disable Trusted Proxy Support", dest="tproxy_enabled")
+
```

Comment:
Maybe allow for one proxy user to be set up via the CLI args?
- `--tproxy-proxy-user`
- `--tproxy-proxyuser-hosts`
- `--tproxy-proxyuser-users`
- `--tproxy-proxyuser-groups`

Example:
```
ambari-server setup-tproxy -ambari-admin-username=admin --tproxy-enabled=true --tproxy-proxy-user=knox --tproxy-proxyuser-hosts="c7401.ambari.apache.org" --tproxy-proxyuser-users="*" --tproxy-proxyuser-users="users"
```

Reply:
I'm OK with this; but we may go further. What if we supported the following:

```
--tproxy-PROXYUSER-hosts
--tproxy-PROXYUSER-users
--tproxy-PROXYUSER-groups
```

And we would check if there is at least any of `proxy-X-[hosts|users|groups]` is defined, then we should have the rest (i.e `--proxy-X-hosts` and `proxy-X-groups` are defined but there is no `proxy-X-users` then we fail the operation).

Moreover: if there is at least one full configuration of a proxyuser set using these options there should be no more proxy users asked in interactively

This way we allow end-users to define any number of proxy user configurations without user interaction (thus this tool would be 100% scriptable).

Reply:
@rlevas 

As discussed over Zoom the above recommended approach does not really work due to CLI option limitations.  Instead the following CLI option will be defined: `--tproxy-configuration-file-path`; if this is provided a file must exists on the referenced path and must contain the Trusted Proxy configurations in JSON format.

For instance:
```
--tproxy-configuration-file-path=/my/path/tproxy.configs

$ cat /my/path/tproxy.configs
[
  {
    "name"   : "knox",
    "hosts"  : "host1"
    "users"  : "user1"
    "groups" : "group1"
  },
  {
    "name"   : "admimn",
    "hosts"  : "host2"
    "users"  : "user2, user3"
    "groups" : "group2"
  }
]
```

Reply:
Nice... but maybe change `name` to `proxyuser`?